sdk/storage/src/datalake/path_client.cpp

```diff
@@ -328,63 +227,16 @@ namespace Azure { namespace Storage { namespace DataLake {
-
-    DataLakeRestClient::Path::CreateOptions protocolLayerOptions;
-    // TODO: Add null check here when Nullable<T> is supported
-    protocolLayerOptions.Continuation = options.Continuation;
-    protocolLayerOptions.Mode = options.Mode;
-    protocolLayerOptions.SourceLeaseId = options.SourceAccessConditions.LeaseId;
-    protocolLayerOptions.LeaseIdOptional = options.AccessConditions.LeaseId;
-    protocolLayerOptions.IfMatch = options.AccessConditions.IfMatch;
-    protocolLayerOptions.IfNoneMatch = options.AccessConditions.IfNoneMatch;
-    protocolLayerOptions.IfModifiedSince = options.AccessConditions.IfModifiedSince;
-    protocolLayerOptions.IfUnmodifiedSince = options.AccessConditions.IfUnmodifiedSince;
-    protocolLayerOptions.SourceIfMatch = options.SourceAccessConditions.IfMatch;
-    protocolLayerOptions.SourceIfNoneMatch = options.SourceAccessConditions.IfNoneMatch;
-    protocolLayerOptions.SourceIfModifiedSince = options.SourceAccessConditions.IfModifiedSince;
-    protocolLayerOptions.SourceIfUnmodifiedSince = options.SourceAccessConditions.IfUnmodifiedSince;
-    protocolLayerOptions.RenameSource = "/" + m_dfsUri.GetPath();
-    protocolLayerOptions.Timeout = options.Timeout;
     auto result = DataLakeRestClient::Path::Create(
-        destinationDfsUri.ToString(), *m_pipeline, options.Context, protocolLayerOptions);
-    // At this point, there is not more exception thrown, meaning the rename is successful.
-    m_dfsUri = std::move(destinationDfsUri);
-    m_blobUri = Details::GetBlobUriFromDfsUri(m_dfsUri);
-    return result;
-  }
-
-  PathDeleteResponse PathClient::Delete(const PathDeleteOptions& options) const
-  {
-    DataLakeRestClient::Path::DeleteOptions protocolLayerOptions;
-    // TODO: Add null check here when Nullable<T> is supported
-    protocolLayerOptions.Continuation = options.Continuation;
-    protocolLayerOptions.LeaseIdOptional = options.AccessConditions.LeaseId;
-    protocolLayerOptions.IfMatch = options.AccessConditions.IfMatch;
-    protocolLayerOptions.IfNoneMatch = options.AccessConditions.IfNoneMatch;
-    protocolLayerOptions.IfModifiedSince = options.AccessConditions.IfModifiedSince;
-    protocolLayerOptions.IfUnmodifiedSince = options.AccessConditions.IfUnmodifiedSince;
-    protocolLayerOptions.RecursiveOptional = options.RecursiveOptional;
-    protocolLayerOptions.Timeout = options.Timeout;
-    return DataLakeRestClient::Path::Delete(
         m_dfsUri.ToString(), *m_pipeline, options.Context, protocolLayerOptions);
+    auto ret = PathInfo();
+    ret.Date = std::move(result.Date);
+    ret.ETag = std::move(result.ETag);
+    ret.LastModified = std::move(result.LastModified);
+    ret.RequestId = std::move(result.RequestId);
+    ret.Version = std::move(result.Version);
+    ret.ContentLength = std::move(result.ContentLength);
+    return ret;
   }
 
   GetPathPropertiesResponse PathClient::GetProperties(const PathGetPropertiesOptions& options) const
```

Comment:
Are we going to have a `GetAccessControl` or not?

Reply:
Talked offline, added.